sdks/python/apache_beam/io/mongodbio.py

```diff
@@ -241,6 +275,27 @@ def _get_split_keys(self, desired_chunk_size_in_mb, start_pos, end_pos):
               max={'_id': end_pos},
               maxChunkSize=desired_chunk_size_in_mb)['splitKeys'])
 
+  def _get_buckets(self, desired_chunk_size, start_pos, end_pos):
+    if start_pos >= end_pos:
+      # single document not splittable
+      return []
+    size = self.estimate_size()
+    bucket_count = size // desired_chunk_size
+    if size % desired_chunk_size != 0:
+      bucket_count += 1
+    with beam.io.mongodbio.MongoClient(self.uri, **self.spec) as client:
+      buckets = list(
```

Comment:
the return buckets should guarantee the _id range is start_pos and end_pos otherwise same document could be read multiple times.

Reply:
Fixed - return buckets cover all the requested range.